splunk_eventgen/lib/eventgentimer.py

```diff
@@ -32,7 +32,7 @@ def __init__(self, time, sample=None, config=None, genqueue=None, outputqueue=No
         self.profiler = config.profiler
         self.config = config
         self.sample = sample
-        self.end = getattr(self.sample, "end") if getattr(self.sample, "end") is not None else -1
```

Comment:
This is introduced in this pr: https://github.com/splunk/eventgen/pull/189/files.  `getattr` will throw exception when `end` is not defined in `self.sample`.